### PR TITLE
github-action: run buildkite action with GH secrets

### PR DESCRIPTION
**Pull request by v1v**
Use GitHub secrets to run the Buildkite release/snapshot

No more Vault secrets access.

There is a pending task to replace the GitHub token user.

### Tests

I created a feature branch called `test/buildkite-token` and triggered a dry-run snapshot:
- https://github.com/elastic/ecs-logging-java/actions/runs/9365973088

### CODE REVIEW

**Comment by SylvainJuge** [comment]
FYI I've renamed the `release.yml` file in https://github.com/elastic/ecs-logging-java/pull/260, so if it's merged before this one you will likely have to double-check that your changes are included in the `release-step-3.yml`.

**Review by SylvainJuge — APPROVED** [review]

File: .github/workflows/snapshot.yml [hunk 1]
```diff
@@ -73,7 +71,7 @@ jobs:
           subject-path: "${{ github.workspace }}/**/target/*.jar"
 
       - if: ${{ failure() }}
-        uses: elastic/oblt-actions/slack/send@v1.2.0
+        uses: elastic/oblt-actions/slack/send@v1.5.0
```
Comment (reakaleek) [inline .github/workflows/snapshot.yml:74]: Maybe we can use `@v1` for actions  where we are sure that it won't have any effect on the build output.

Comment (reakaleek) [reply .github/workflows/snapshot.yml:74]: but if it's easier to just use the same version ref. than let's stick with it.

Comment (v1v) [reply .github/workflows/snapshot.yml:74]: I don't know yet.

If we need to do something to all the repositories we could use the `gh oblt` extension and apply the sam pattern.

For now, let's use the latest available version, and let dependabot drive the bumps

**Pull request merged**